js/src/Mark.ts

```diff
@@ -197,13 +197,15 @@ export abstract class Mark extends widgets.WidgetView {
         this.d3el.style("display", visible ? "inline" : "none");
     }
 
-    get_colors(index) {
+    //how do I know if smth is a color attr or a data attr, this changes the type
+    //cause it could be a numpy array
+    get_colors(index): string {
```

Comment:
I'm not super sure about this type. It comes from `color` from here: https://github.com/bqplot/bqplot/blob/5595c241093d6f300eac76795d03217c625922bc/bqplot/marks.py#L368, but I'm confused... It just says that's an array.
Maybe it's even just a `: Color` type, because we're getting it from an array of colors? (https://github.com/bqplot/bqplot/blob/5595c241093d6f300eac76795d03217c625922bc/bqplot/marks.py#L304)

Reply:
There are two parameters: `color` and `colors`.
- `color` is an array of numbers, and we use the color scale to map this number to an actual color string. To give you an example, this is useful when you want to map some data to a color (like mapping a stock price to a bar color in a bar chart).
- `colors` is a list of string (CSS colors). Those are the "default" colors that are used either when the color data is NaN or when there are no color data (if `color` is None).

You can find an example of this in the `examples/Marks/Pyplot/Lines.ipynb` Notebook, section "Coloring Lines according to data".

This is not true for all the marks, but it is at least true for Lines, Bar, Scatter and I think PieChart.

So basically you are right, the return type is a string here.
You can also type `index` as being a `number` here: `get_colors(index: number): string {`